src/actions/events.js

```diff
@@ -405,6 +495,34 @@ events.on(FILTER_CLEAR, function() {
   tree.unset(['pages', 'repo', 'filter']);
 });
 
+events.on(BUILD_FILTER, function(event) {
+  const {owner, name, value} = event.data;
+  if (value === '') {
+    tree.unset(['pages', 'repo', owner, name, 'filter']);
+  } else {
+    tree.set(['pages', 'repo', owner, name, 'filter'], value);
+  }
```

Comment:
@bradrydzewski Updated. It resets filters when you leave repo page. If there is any more feedback on that please let me know. If all good I'll squash commits and change PR name to be no longer WIP.